### ISSUE TITLE
Date issues under firefox

### ISSUE THREAD

**Issue by serjrd** [acidb/mobiscroll#300]
Thanks for you work guys, I find your plugin really useful.

Not sure what exactly is the source of the problem, but under the current version of Firefox developer edition 39.0a2 (2015-04-08) there's a problem with the mobiscroll date demo.

On your [demo page](http://demo.mobiscroll.com/datetime/dateofbirth/#?device=np&display=inline) if I choose March, 1981-1984 the only available day number is 1.
No errors are reported to console.

**Comment by serjrd** [comment]
Just checked it under a regular Firefox 37.0.1. Same issue.

**Comment by dioslaska** [comment]
Hi, I was not able to reproduce it under Firefox 37.0.1 (desktop and mobile). It could be related with #295 , but I could not reproduce it by changing the timezone. 
Could you please provide your timezone settings?

**Comment by serjrd** [comment]
My timezone is GMT+3.

I don't know if it's relevant, but both versions of Firefox that I tried this on, were run on Linux.
Under Chrome there's no such problem.

**Comment by dioslaska** [comment]
Unfortunately I'm still not able to reproduce it.
Using Ubuntu 14.10, Firefox 37.0.1.

What is your exact timezone (inlcuding city)? The other bug only occures if timezone is explicitly GMT London, other locations with GMT+0 are working fine.

**Comment by serjrd** [comment]
It's GMT+3/Moscow.
I tried running `new Date(1984,2,1)` in the Firefox console:

```
new Date(1984,2,1)
Date 1984-02-29T21:00:00.000Z
new Date(1984,2,2)
Date 1984-03-01T21:00:00.000Z
```

The result looks correct, given that it's shown in the Zulu timezone (GTM).

Of course, I don't know the details, but do you actually need to use Date() to figure out the possible dates and time? The number of days is always the same (except for the leap years, but they are really easy to calculate) - so with just a very simple code you can display the right dates no matter what bug are there in the Date() implementation. At least that's what I've ended up doing.

**Comment by dioslaska** [comment]
Ok, I was finally able to reproduce it. Explicitly with the Moscow timezone and only on Linux.
The problem is with this line:

    new Date(1981, 2, 32).getDate()

This should return 1, but it returns 31. It is used to determine the number of days in a given month:

    32 - new Date(y, m, 32).getDate()

You are right, this could be done without using javascript Date, but with Date is much easier to work with. Besides the above case we also using Date objects for validation (min, max, and invalid days), which would be much harder to write with other methods. 

For now I will file a bug report to Mozilla.

**Comment by serjrd** [comment]
Oh my.
Glad you've managed to track down the issue.

Thanks again for all the great work you're doing!

**Comment by dioslaska** [comment]
Mozilla bug report: https://bugzilla.mozilla.org/show_bug.cgi?id=1155096

**Comment by dioslaska** [comment]
Merged into #295
